itests/hive-unit/src/test/java/org/apache/hive/jdbc/TestSSL.java

```diff
@@ -437,15 +439,36 @@ public void testConnectionWrongCertCN() throws Exception {
    * Test HMS server with SSL
    * @throws Exception
    */
+  @Ignore
   @Test
   public void testMetastoreWithSSL() throws Exception {
     testSSLHMS(true);
   }
 
+  /**
+   * Test HMS server with Http + SSL
+   * @throws Exception
+   */
+  @Test
+  public void testMetastoreWithHttps() throws Exception {
+    // MetastoreConf.setBoolVar(conf, MetastoreConf.ConfVars.EVENT_DB_NOTIFICATION_API_AUTH, false);
+    //MetastoreConf.setVar(conf, MetastoreConf.ConfVars.METASTORE_CLIENT_TRANSPORT_MODE, "http");
+    SSLTestUtils.setMetastoreHttpsConf(conf);
+    MetastoreConf.setVar(conf, MetastoreConf.ConfVars.SSL_TRUSTMANAGERFACTORY_ALGORITHM,
+        KEY_MANAGER_FACTORY_ALGORITHM);
+    MetastoreConf.setVar(conf, MetastoreConf.ConfVars.SSL_TRUSTSTORE_TYPE, KEY_STORE_TRUST_STORE_TYPE);
+    MetastoreConf.setVar(conf, MetastoreConf.ConfVars.SSL_KEYSTORE_TYPE, KEY_STORE_TRUST_STORE_TYPE);
+    MetastoreConf.setVar(conf, MetastoreConf.ConfVars.SSL_KEYMANAGERFACTORY_ALGORITHM,
+        KEY_MANAGER_FACTORY_ALGORITHM);
+
+    testSSLHMS(false);
```

Comment:
Why are we passing false here? This value is used in testSSLHMS()#L459-461 to set the keystore for HMS and HS2. You are already setting this for HMS in L461 here and we don't need to set for HS2. So why don't we just pass the value true?

Reply:
Thanks for pointing it out. I am setting the conf `MetastoreConf.ConfVars.SSL_KEYSTORE_TYPE`  in testSSLHMS(false) now.